From 9657d54d892d45438c6dcdee6361426feb7118f1 Mon Sep 17 00:00:00 2001
From: Son Lam <son.lam.ym@rvc.renesas.com>
Date: Thu, 1 Aug 2019 11:45:58 +0700
Subject: [PATCH] weston: add weston_desktop_api_set_xwayland_position

Description: wayland applications expect - geometry command line option
to work for setting the initial window position, but currently this
does not work.

Call provision to relay the initial position through libweston-desktop:
- weston_desktop_api gains a new entry set_xwayland_position

Once wl-shell plugin starts calling
weston_desktop_wl_shell_surface_protocol_set_transient(),
libweston-desktop can relay that information to a shell if the shell
wants to hear it.

If a shell does not implement the set_xwayland_position hook, the
old behaviour remains: the shell positions wl-shell toplevels like any
other.

Signed-off-by: Pekka Paalanen <pekka.paalanen at collabora.co.uk>
Signed-off-by: Thang Tran <ThangT@fsoft.com.vn>

Signed-off-by: Son Lam <son.lam.ym@rvc.renesas.com>
---
 libweston-desktop/wl-shell.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libweston-desktop/wl-shell.c b/libweston-desktop/wl-shell.c
index 399139c..4fa3ec9 100644
--- a/libweston-desktop/wl-shell.c
+++ b/libweston-desktop/wl-shell.c
@@ -286,6 +286,9 @@ weston_desktop_wl_shell_surface_protocol_set_transient(struct wl_client *wl_clie
 		weston_desktop_api_set_parent(surface->desktop,
 					      surface->surface, parent);
 	}
+
+	weston_desktop_api_set_xwayland_position(surface->desktop, surface->surface,
+                                                    x, y);
 }
 
 static void
-- 
2.7.4

